crates/relayer-types/src/core/ics04_channel/channel.rs

```diff
@@ -498,28 +489,28 @@ impl State {
 
```

Comment:
This method is becoming unwieldy and complex, and we should either:

a) write good unit tests for it, to ensure that it's correct but also that it is isomorphic to `self.as_i32() <= other.as_i32()`.
b) express it in terms of `as_i32()`

Given that doing (a) enables (b), I think we should do (b) right away.


@ljoss17 What do you think?

Reply:
I agree, this would be a nice improvement. I will add the unit test/s and update the method